assets/js/app/Data.js

```diff
@@ -175,30 +178,94 @@ LocusZoom.Data.AssociationSource.prototype.parseResponse = function(resp, chain,
         }
         records.push(record);
     }
-    var res = {header: chain.header || {}, body: records};
-    return res;
+    return records;
+};
+
+LocusZoom.Data.Source.prototype.parseObjectsToObjects = function(x, fields, outnames, trans) {
+    //intended for an array of objects
+    // [ {"id":1, "val":5}, {"id":2, "val":10}]
+    var records = [];
+    var fieldFound = [];
+    for (var i = 0; i < x.length; i++) {
+        var record = {};
+        for (var j=0; j<fields.length; j++) {
+            var val = x[i][fields[j]];
+            if (typeof val != "undefined") {
```

Comment:
Good point. Actually. If `c` isn't found, it should be `[1,1,undefined,1]` but `forEach` seems to skip over undefined values. It would have been even worse if the missing field(s) was(were) at the end of the list. I'll go ahead and initialize all to 0.